### PR TITLE
Throw when System.Transactions.Transaction.Current is set

### LINKED ISSUE #1584: Throw when System.Transactions.Transaction.Current is set
When discussing https://github.com/aspnet/EntityFramework/issues/1530 we decided not to support TransactionScope. We decided we should detect if someone is trying to though and throw so it is clear it's not supported. We should have a flag to disable throwing in case folks are using TransactionScope for other reasons and EF code just happens to be hit in the block.

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Fixes #1584

### CODE REVIEW

**Comment by AndriySvyryd** [comment]
@ajcvickers, @divega, @rowanmiller

File: src/EntityFramework.SqlServer/Extensions/SqlServerDbContextOptions.cs [hunk 1]
```diff
@@ -39,6 +39,18 @@ public virtual SqlServerDbContextOptionsBuilder CommandTimeout(int? commandTimeo
             return this;
         }
 
+        public virtual SqlServerDbContextOptionsBuilder SuppressAmbientTransactionWarning()
+        {
+            var extension = new SqlServerOptionsExtension(OptionsBuilder.Options.FindExtension<SqlServerOptionsExtension>());
+            Debug.Assert(extension != null);
```
Comment (ajcvickers) [inline src/EntityFramework.SqlServer/Extensions/SqlServerDbContextOptions.cs]: Consider refactoring into a helper method and moving the assertion, which I think should be asserting that FindExtension is not returning null.

File: src/EntityFramework.Relational/RelationalConnection.cs [hunk 2]
```diff
@@ -153,19 +160,47 @@ public virtual RelationalTransaction UseTransaction([CanBeNull] DbTransaction tr
...
         }
 
         public virtual async Task OpenAsync(CancellationToken cancellationToken = default(CancellationToken))
         {
+#if NET451
+            if(_throwOnAmbientTransaction
+               && System.Transactions.Transaction.Current != null)
+            {
+                throw new InvalidOperationException(Strings.AmbientTransaction);
+            }
```
Comment (ajcvickers) [inline src/EntityFramework.Relational/RelationalConnection.cs]: Consider DRY into a helper method.

**Comment by ajcvickers** [comment]
:shipit:

**Pull request merged**